### ISSUE TITLE
Opening CogTool on Mac without JRE 1.5 - problem and partial solution

### ISSUE THREAD

**Issue by azmr** [cogtool/cogtool#3]
(I'm not particularly familiar with the Java ecosystem, wherever I mention the 'Java Runtime Environment' (JRE), the 'Java Development Kit' (JDK) may be more appropriate)

Problem
======
TL;DR:
- CogTool assumes JRE 1.5, which can't be installed on the latest Macs (and the latest JRE is not compatible).

I had Java Runtime Environment (JRE) 9 installed on my High Sierra Mac, but when I tried to open CogTool nothing happened: the icon would bounce a couple of times to say it was opening, then disappear.

If you explore the contents of the app (right-click and select "Show Package Contents") then open `Contents/MacOS/cogtoolstart`, you can see that opening the app tries java 1.5, then if that doesn't exist, just tries the system java. In my case (v9), this didn't work. If you double click `cogtoolstart`, a Terminal window opens showing the usage instructions for `java`, which you'd expect might happen if opening the wrong version.

Solution
======
TL;DR:
- Download [JRE 1.6 for Mac](https://support.apple.com/kb/DL1572?viewlocale=en_GB&locale=en_GB)
- Edit `/Applications/CogTool/Contents/MacOS/cogtoolstart` so that `JAVACMD="/path/to/your/java"` (e.g. `JAVACMD="/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/bin/java"`)

I had some difficulty finding a JRE 1.5 that worked for my version of Mac. It seems like it might have been deprecated?
After a bit of searching I found that JRE 1.6 (aka JRE 6) is basically backwards compatible with 1.5, and there's a version that installs on my machine: ["Download Java for OS X 2017-001 "](https://support.apple.com/kb/DL1572?viewlocale=en_GB&locale=en_GB).

The next issue to solve was that CogTool was trying to open the default java (JRE 9 for me), which I didn't want to overwrite, as it's used elsewhere. Simply enough, `JAVACMD` in `cogtoolstart` could be updated to the new java location, which for me was `JAVACMD="/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/bin/java"`.

After this, I could open CogTool from the Applications folder as normal. This fix seems to work for both the stable and development versions of CogTool

**Comment by azmr** [comment]
I spoke too soon saying that using 1.6 was a fix. It looks like the SWT version is incompatible. Below is the error message that occurs when I click 'Show Visualization' for a task. After a bit of trial-and-error, this seemed to fix the problem:
- I found a [legacy version of SWT from a previous version of Eclipse (3.5.2)](http://archive.eclipse.org/eclipse/downloads/drops/R-3.5.2-201002111343/) that is compatible. Scroll down to the bottom and download the lowest 32-bit `swt-3.5.2-carbon-macosx.zip` from 'SWT Binary and Source'.
- Rename `/Applications/CogTool/Contents/Resources/Java/swt.jar` so you don't lose it. e.g. to `swtbackup.jar`.
- Copy in `swt.jar` from the folder just downloaded from Eclipse
- Open CogTool as normal

Error:
```
Version: 1.2.2.0 (3788) 01/10/2013-09:42
Mac OS X, version 10.13.3, JVM 1.6.0_65 (Apple Inc.)
Memory usage: 53.78% of 126.99;  395.58
java.lang.UnsatisfiedLinkError: no swt-cocoa-carbon-3448 or swt-cocoa-carbon in swt.library.path, java.library.path or the jar file
	at org.eclipse.swt.internal.Library.loadLibrary(Unknown Source)
	at org.eclipse.swt.internal.Library.loadLibrary(Unknown Source)
	at org.eclipse.swt.internal.cocoa.Cocoa.<clinit>(Unknown Source)
	at org.eclipse.swt.graphics.Cursor.createNSCursor(Unknown Source)
	at org.eclipse.swt.graphics.Cursor.<init>(Unknown Source)
	at edu.cmu.cs.hcii.cogtool.util.WindowUtil.getOpenHandCursor(WindowUtil.java:179)
	at edu.cmu.cs.hcii.cogtool.view.PERTScrollBar.<init>(PERTScrollBar.java:315)
	at edu.cmu.cs.hcii.cogtool.view.PERTChartPanel.<init>(PERTChartPanel.java:205)
	at edu.cmu.cs.hcii.cogtool.view.PERTChartView.<init>(PERTChartView.java:315)
	at edu.cmu.cs.hcii.cogtool.ui.PERTChartUI.<init>(PERTChartUI.java:175)
	at edu.cmu.cs.hcii.cogtool.controller.PERTChartController.<init>(PERTChartController.java:121)
	at edu.cmu.cs.hcii.cogtool.controller.PERTChartController.openController(PERTChartController.java:202)
	at edu.cmu.cs.hcii.cogtool.controller.SEDemoController$27.performAction(SEDemoController.java:1851)
	at edu.cmu.cs.hcii.cogtool.util.ListenerIdentifierMap.performAction(ListenerIdentifierMap.java:193)
	at edu.cmu.cs.hcii.cogtool.util.ListenerIdentifierMap.performAction(ListenerIdentifierMap.java:244)
	at edu.cmu.cs.hcii.cogtool.ui.UI.performAction(UI.java:559)
	at edu.cmu.cs.hcii.cogtool.ui.UI.performAction(UI.java:529)
	at edu.cmu.cs.hcii.cogtool.ui.DefaultUI.performAction(DefaultUI.java:477)
	at edu.cmu.cs.hcii.cogtool.view.View.performAction(View.java:602)
	at edu.cmu.cs.hcii.cogtool.view.View$SWTWidgetChangeHandler.widgetSelected(View.java:174)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(Unknown Source)
	at org.eclipse.swt.widgets.EventTable.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Widget.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Widget.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Widget.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Unknown Source)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Unknown Source)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Unknown Source)
	at edu.cmu.cs.hcii.cogtool.util.WindowUtil.interact(WindowUtil.java:517)
	at edu.cmu.cs.hcii.cogtool.CogTool.main(CogTool.java:398)
```

**Comment by justingeeslin** [comment]
The SWT issue show above is fixed by PR #12  .
